woocommerce/woocommerce
# [Experimental]: Create Variation Selector Block in editor

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Partially addresses [53012](https://github.com/woocommerce/woocommerce/issues/53012). Second part [here](https://github.com/woocommerce/woocommerce/pull/53947).

This PR adds the code to include the variation selector in the editor. The change is part of [a bigger PR](https://github.com/woocommerce/woocommerce/pull/53855) split to make reviewing easier.

![Screenshot 2024-12-20 at 5 09 55 PM](https://github.com/user-attachments/assets/5a73913c-3cab-40fc-8439-d66ef1d938b5)

The changes in this PR will be visible only in the editor, not in the front end.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Having `WooCommerce Beta Tester` installed and activater
1. Go to Tools > WCA Test Helper > Features
2. Enable the features: `blockified-add-to-cart` and `experimental-blocks`
3. Create a variable product with variations.
4. Go to Pages > Add New Page
5. Add a `Single Product` block and select the product you created in step 3.
6. Add an `Add to Cart with Option (Experimental)` block.
7. Verify that the selectors with the attributes are visible.
8. Now change the product and choose a simple one
9. The selector should disappear.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
